src/wallet/api/pending_transaction.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2014-2016, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Parts of this file are originally copyright (c) 2012-2013 The Cryptonote developers
+
+#include "wallet/wallet2_api.h"
+#include "wallet/wallet2.h"
+
+#include <string>
+#include <vector>
+
+
+namespace Bitmonero {
+
+class WalletImpl;
+class PendingTransactionImpl : public PendingTransaction
+{
+public:
+    PendingTransactionImpl(WalletImpl * wallet);
+    ~PendingTransactionImpl();
+    int status() const;
+    std::string errorString() const;
+    bool commit();
+    uint64_t amount() const;
+    uint64_t dust() const;
+    uint64_t fee() const;
+    // TODO: continue with interface;
+
+private:
+    friend class WalletImpl;
+    WalletImpl * m_wallet;
```

Comment:
I tend to prefer references for the non-null guarantee, as some other
developer may carelessly pass a null pointer to the constructor of any
arbitrary class such as this in the future.

On Wed, Jun 22, 2016 at 8:03 AM, Ilya Kitaev <notifications@github.com>
wrote:

> In src/wallet/api/pending_transaction.h
> <https://github.com/monero-project/bitmonero/pull/872#discussion_r68040378>
> :
>
> > +class PendingTransactionImpl : public PendingTransaction
> > +{
> > +public:
> > +    PendingTransactionImpl(WalletImpl * wallet);
> > +    ~PendingTransactionImpl();
> > +    int status() const;
> > +    std::string errorString() const;
> > +    bool commit();
> > +    uint64_t amount() const;
> > +    uint64_t dust() const;
> > +    uint64_t fee() const;
> > +    // TODO: continue with interface;
> > +
> > +private:
> > +    friend class WalletImpl;
> > +    WalletImpl * m_wallet;
>
> No special reasons for using pointers vs references here.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/monero-project/bitmonero/pull/872/files/eec0f57d60a8c2f69b18566b2baecfb65a2f641d#r68040378>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE3k5rdsxncn1WakKfwXgGsF_X2WH-wWks5qOSSSgaJpZM4I6mUi>
> .
>



-- 
Thomas Winget
Computer Engineering
Purdue University '12